lib/utils/mobile.js

```diff
@@ -294,6 +294,12 @@ class IosSessionNotCreatedError extends Error {
           `For more help, run: ${Logger.colors.cyan('npx run @nightwatch/mobile-helper ios')}\n`
         ];
       }
+    } else if (this.desiredCapabilities['safari:deviceUDID']) {
+      help = [
+        'Re-run the test command',
+        'Verify device connection and nightwatch configuration',
+        `If it doesn't work, try running: ${Logger.colors.cyan('npx run @nightwatch/mobile-helper ios')}`
+      ];
```

Comment:
Can you please make these error messages a little more detailed? Like the second one is really confusing to me as in what exactly do we need to do?

Also, `safari:deviceUDID` can also be set for iOS Simulators, right? So, this error message cannot be specific to real iOS devices (if that's what's meant here). A comment at the top of this `else if` block would also help to understand in what situations users might get this help message, just like you've added in the above `if/else if` blocks.

Reply:
The help message is still not clear, since it does not mention where to look for the device id in the Nightwatch configuration. Maybe change it to:

Verify the UDID of the device set in Nightwatch configuration (look for `safari:deviceUDID` desired capability) or passed using the `--deviceId` flag.

Reply:
Have a look once [48e2a4f](https://github.com/nightwatchjs/nightwatch/pull/3572/commits/48e2a4fc2992c55810603d0bbef701532c0b1661)